multitenancy/tenant-manager-api/src/main/java/io/apicurio/multitenant/auth/CustomAuthenticationMechanism.java

```diff
@@ -0,0 +1,123 @@
+/*
+ * Copyright 2021 Red Hat
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.apicurio.multitenant.auth;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.BiConsumer;
+
+import javax.annotation.Priority;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Alternative;
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+
+import io.apicurio.multitenant.logging.audit.AuditHttpRequestContext;
+import io.apicurio.multitenant.logging.audit.AuditHttpRequestInfo;
+import io.apicurio.multitenant.logging.audit.AuditLogService;
+import io.quarkus.oidc.runtime.BearerAuthenticationMechanism;
+import io.quarkus.oidc.runtime.OidcAuthenticationMechanism;
+import io.quarkus.security.identity.IdentityProviderManager;
+import io.quarkus.security.identity.SecurityIdentity;
+import io.quarkus.security.identity.request.AuthenticationRequest;
+import io.quarkus.security.identity.request.TokenAuthenticationRequest;
+import io.quarkus.vertx.http.runtime.security.ChallengeData;
+import io.quarkus.vertx.http.runtime.security.HttpAuthenticationMechanism;
+import io.quarkus.vertx.http.runtime.security.HttpCredentialTransport;
+import io.quarkus.vertx.http.runtime.security.QuarkusHttpUser;
+import io.smallrye.mutiny.Uni;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * Custom HttpAuthenticationMechanism that simply wraps OidcAuthenticationMechanism.
+ * The only purpose of this HttpAuthenticationMechanism is to handle authentication errors in order to generate audit logs.
+ *
+ * @author Fabian Martinez
+ */
+@Alternative
```

Comment:
Yes, I forgot this is required, I added it in Registry to the BasicAuthClientCredentialsMechanism and that one is relying on the CustomAuthenticationMechanism, sorry about that.